### PR TITLE
[SYCL][HIP] Fix clang warnings

### PR DESCRIPTION
**Pull request by t4c1**
Fixes compiler warnings when compiling for the ROCm backend using clang compiler. Since `-Werror` is used by default, these warnings prevent the build from succeeding.

### CODE REVIEW

**Comment by bader** [comment]
@smaslov-intel, ping.

File: sycl/plugins/rocm/pi_rocm.cpp [hunk 1]
```diff
@@ -1639,7 +1655,8 @@ pi_result rocm_piContextCreate(const pi_context_properties *properties,
       break;
     default:
       // Unknown property.
-      assert(!"Unknown piContextCreate property in property list");
+      assert(!static_cast<bool>(
```
Comment (smaslov-intel) [inline sycl/plugins/rocm/pi_rocm.cpp:1658]: this doesn't look pretty, and what is the cast of char* to bool doing? also assert will not fire in product (built without assertions)

Comment (smaslov-intel) [reply sycl/plugins/rocm/pi_rocm.cpp:1658]: so maybe just use "die" here?

**Review by smaslov-intel — APPROVED** [review]
Just one small correction suggested

File: sycl/plugins/rocm/pi_rocm.cpp (hunk 1 above)
Comment (t4c1) [reply sycl/plugins/rocm/pi_rocm.cpp:1658]: Negating char* gives a warning about implicit conversion to bool (regardless of product/debug). So I instead made the cast explicit. I am not sure what do you mean with "die".

Comment (smaslov-intel) [reply sycl/plugins/rocm/pi_rocm.cpp:1658]: see line 870 in the same file

**Comment by bader** [comment]
@t4c1, please, update your branch to resolve conflicts.

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
> Just one small correction suggested

@smaslov-intel, does it look okay now?

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**